### DISCUSSION TITLE
Config merging

### DISCUSSION THREAD

**Discussion by kotomkin** [laravel/framework#51818] [Q&A] [answered]
Laravel 11 is merging config from `project/config` directory with config from `laravel/framework/config` directory. The merge works for top-level options and for some nested options that are defined in `laravel/framework/src/Illuminate/Foundation/Bootstrap/LoadConfiguration.php` file:

![image](https://github.com/laravel/framework/assets/34843587/7ab48ed4-6835-4a43-bf22-45030d649f1e)

This includes the merging of `'database.connections'` option. For example, if I will remove `'connections.mysql'` option from my `project/config/database.php` file, Laravel will take its value from `laravel/framework/config/database.php` file over which I have no control.
`project/config/database.php`:

![image](https://github.com/laravel/framework/assets/34843587/4246a3d7-bef3-4a1b-b494-d15fc9b2ee07)
result:
![image](https://github.com/laravel/framework/assets/34843587/fc754248-6ca3-4d73-a76c-4b4116e0445b)

In other words, I can't remove unnecessary connections from the final config. I would like to have more control over my final config. Is there any mechanism to change the above behavior?

**Comment by kotomkin** [comment, marked as answer]
Ok, I found `Illuminate\Foundation\Application::dontMergeBaseConfiguration()` method.
